### PR TITLE
feat(template): Make kebab-case filter working with digits in the middle of a word

### PR DESCRIPTION
**Pull request by lquerel**
"K8S" | kebab-case  => "k8s"

### CODE REVIEW

**Comment by jsuereth** [comment]
FYI - I found this issue is the same for  the `TitleCase` filter too.

File: crates/weaver_forge/src/lib.rs [hunk 1]
```diff
@@ -355,19 +354,19 @@ impl TemplateEngine {
             "field_name",
             case_converter(self.target_config.field_name.clone()),
         );
-        env.add_filter("lowercase", case_converter(CaseConvention::LowerCase));
-        env.add_filter("UPPERCASE", case_converter(CaseConvention::UpperCase));
-        env.add_filter("TitleCase", case_converter(CaseConvention::TitleCase));
-        env.add_filter("PascalCase", case_converter(CaseConvention::PascalCase));
-        env.add_filter("camelCase", case_converter(CaseConvention::CamelCase));
+        env.add_filter("lower_case", case_converter(CaseConvention::LowerCase));
```
Comment (jsuereth) [inline crates/weaver_forge/src/lib.rs:357]: This is much nicer to use! thanks.

File: crates/weaver_forge/src/config.rs [hunk 2]
```diff
@@ -304,7 +308,22 @@ impl CaseConvention {
...
             CaseConvention::ScreamingSnakeCase => text.to_case(Case::ScreamingSnake),
-            CaseConvention::KebabCase => text.to_case(Case::Kebab),
+            CaseConvention::KebabCase => {
+                // Convert to kebab case but do not consider digits
+                // as boundaries. So that `k8s` will stay `k8s` and
+                // not `k-8-s`.
+                let conv = KEBAB_CASE.get_or_init(|| {
+                    Converter::new()
+                        .add_boundary(Hyphen)
+                        .remove_boundary(DigitLower)
```
Comment (jsuereth) [inline crates/weaver_forge/src/config.rs:318]: Could you add this fix to LowerCase + TitleCase too?

File: crates/weaver_forge/templates/test/converter.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{%- set text = "This is a TEST" -%}
```
Comment (jsuereth) [inline crates/weaver_forge/templates/test/converter.md:1]: Maybe put `k8s` as a word in here to see what happens.

File: crates/weaver_forge/src/config.rs (hunk 2 above)
Comment (lquerel) [reply crates/weaver_forge/src/config.rs:349]: Done -> https://github.com/open-telemetry/weaver/pull/106/commits/afdf547f705fae69d5b4c751fdb3998100d352a9

File: crates/weaver_forge/templates/test/converter.md (hunk 3 above)
Comment (lquerel) [reply crates/weaver_forge/templates/test/converter.md:1]: Done

**Comment by lquerel** [comment]
@jsuereth Please review. That should be ok now.

**Review by jsuereth — APPROVED** [review]

**Pull request merged**